### ISSUE TITLE
[Issue]: [10.8.0] Permission issues running as root

### ISSUE THREAD

**Issue by SingingFrog7** [jellyfin/jellyfin#7897]
### Please describe your bug

I know it may not be a good practice to run Jellyfin as root (even if it runs in a dedicated container) but I wanted to use ports 80 and 443 without adding a reverse-proxy

With Jellyfin 10.7.0, on Debian, one could simply edit `/etc/systemd/system/jellyfin.service.d/jellyfin.service.conf` to modify the user, but that doesn't seem to work anymore. The executable is still ran by root (as shown by `ps -aux`) but doesn't have permissions on files or permission to run on port < 1024.

I am not sure what changed behind the scenes that could cause this?
Manually fixing permissions and running on a port > 1024 works, but I still feel like it's still not really running as root

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS: Debian
- Virtualization: LXC
- Clients: N/A
- Browser: N/A
- FFmpeg Version: 5 
- Playback Method: N/A
- Hardware Acceleration: N/A
- Plugins: N/A
- Reverse Proxy: no
- Base URL: N/A
- Networking: Try to run as root, using port 80 and 443
- Storage:
```

### Jellyfin logs

```shell
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]: [16:43:30] [INF] Kestrel listening on Any IP4 Address
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]: [16:43:30] [ERR] Kestrel failed to start! This is most likely due to an invalid address or port bind - correct your bind configuration in network.xml and try again.
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]: [16:43:30] [FTL] Error while starting server.
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]: System.Net.Sockets.SocketException (13): Permission denied
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at System.Net.Sockets.Socket.Bind(EndPoint localEP)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportOptions.CreateDefaultBoundListenSocket(EndPoint endpoint)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.Bind()
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportFactory.BindAsync(EndPoint endpoint, CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.TransportManager.BindAsync(EndPoint endPoint, ConnectionDelegate connectionDelegate, EndpointConfig endpointConfig, CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.<>c__DisplayClass30_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]: --- End of stack trace from previous location ---
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context, CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions.BindAsync(AddressBindContext context, CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.EndpointsStrategy.BindAsync(AddressBindContext context, CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(IEnumerable`1 listenOptions, AddressBindContext context, CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Microsoft.AspNetCore.Hosting.WebHost.StartAsync(CancellationToken cancellationToken)
Jun 12 16:43:30 lxc-jellyfin jellyfin[961]:    at Jellyfin.Server.Program.StartApp(StartupOptions options)
```
```


### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by joshuaboniface** [comment]
Overriding the user *should* work - verify that with `ps aux` to ensure Jellyfin is actually running as the user you expect.

If it is the issue might be the hardening options (they strike again...). Try the workaround from this comment: https://github.com/jellyfin/jellyfin/issues/7503#issuecomment-1153072208

**Comment by SingingFrog7** [comment]
Thanks for your (really!) quick response.
It did run as root (as shown by `ps -aux`)

Removing `PrivateUsers=true` from the systemd service (or overriding it with `PrivateUsers=false`) seems to fix it

Thanks for your time

**Comment by SingingFrog7** [comment]
I don't know if I am a edge-case user (in which case we may close this bug) or if this needs to be fixed for other users

**Comment by joshuaboniface** [comment]
It will end up being fixed by #7887.
